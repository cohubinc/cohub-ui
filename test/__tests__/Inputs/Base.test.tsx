import React, { ReactNode } from "react";
import { mount } from "enzyme";

import { Inputs, CssFramework, size } from "dist";
import { sleep, getScreenshot } from "../utils";

describe("Base Input", () => {
  describe("image snapshots look correct", async () => {
    it("looks good with no value", async () => {
      const img = await getInputScreenshot(
        <Inputs.Text label="name" input={{ value: "", onChange: () => null }} />
      );

      expect(img).toMatchImageSnapshot();
    });

    it("looks good with a value", async () => {
      const img = await getInputScreenshot(
        <Inputs.Text
          label="name"
          input={{ value: "Brandon", onChange: () => null }}
        />
      );

      expect(img).toMatchImageSnapshot();
    });
  });

  it("renders with value without crashing", async () => {
    const wrapper = mount(<Inputs.Base value="test" onChange={() => null} />);

    expect(wrapper).toMatchSnapshot();
  });
});

async function getInputScreenshot(jsx: ReactNode) {
  return await getScreenshot(jsx, { width: 300, height: 30 });
}
