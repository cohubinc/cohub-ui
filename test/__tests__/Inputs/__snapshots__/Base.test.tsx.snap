// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Base Input renders with value without crashing 1`] = `
<Base
  autoComplete="off"
  autoFocus={false}
  data-qa="base-input-element"
  data-qa-label="base-input-element-label"
  onChange={[Function]}
  type="text"
  value="test"
>
  <FloatingLabelWrapper
    data-qa-label="base-input-element-label"
    onChange={[Function]}
    value="test"
  >
    <div
      className="FloatingLabelWrapper GenericInput "
      style={
        Object {
          "cursor": "text",
          "position": "relative",
        }
      }
    >
      <div
        className="noLabelInputWrapper  bd-radius"
      >
        <input
          autoComplete="off"
          autoFocus={false}
          data-qa="base-input-element"
          data-qa-label="base-input-element-label"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          style={
            Object {
              "color": "hsl(0, 2%, 12%)",
              "cursor": "text",
              "textAlign": undefined,
            }
          }
          type="text"
          value="test"
        />
        <span
          className="bar "
        />
      </div>
    </div>
  </FloatingLabelWrapper>
</Base>
`;

exports[`Base Input renders without crashing 1`] = `
<Base
  autoComplete="off"
  autoFocus={false}
  data-qa="base-input-element"
  data-qa-label="base-input-element-label"
  type="text"
>
  <FloatingLabelWrapper
    data-qa-label="base-input-element-label"
  >
    <div
      className="FloatingLabelWrapper GenericInput "
      style={
        Object {
          "cursor": "text",
          "position": "relative",
        }
      }
    >
      <div
        className="noLabelInputWrapper  bd-radius"
      >
        <input
          autoComplete="off"
          autoFocus={false}
          data-qa="base-input-element"
          data-qa-label="base-input-element-label"
          onBlur={[Function]}
          onFocus={[Function]}
          style={
            Object {
              "color": "hsl(0, 2%, 12%)",
              "cursor": "text",
              "textAlign": undefined,
            }
          }
          type="text"
        />
        <span
          className="bar "
        />
      </div>
    </div>
  </FloatingLabelWrapper>
</Base>
`;
